sky/spot/recovery_strategy.py

```diff
@@ -240,15 +240,16 @@ def _wait_until_job_starts_on_cluster(self) -> Optional[float]:
 
     def _launch(self,
                 max_retry: Optional[int] = 3,
-                raise_on_failure: bool = True) -> Optional[float]:
+                raise_on_max_retry: bool = True) -> Optional[float]:
```

Comment:
Is replacing this arg with a new arg with different semantics ok?

https://github.com/skypilot-org/skypilot/blob/42768949f583380a0551bc05f82685351ec5bd24/sky/spot/recovery_strategy.py#L450 

seems to still use the old arg which eventually gets passed up to this new arg.